### PR TITLE
Fix csr certificate order

### PR DESCRIPTION
**Pull request by baluchicken**
| Q               | A      |
| --------------- | ------ |
| Bug fix?        |yes |
| New feature?    | no|
| API breaks?     | no|
| Deprecations?   | no|
| License         | Apache 2.0 |

### What's in this PR?

This PR changes the certificate chain parsing order, earlier it used a false assumption that the root cert is the last one and the leaf cert is the first one since it is the opposite.

- [X] Implementation tested
- [X] Error handling code meets the [guideline](https://github.com/banzaicloud/developer-guide/blob/master/docs/coding-style/error-handling-guide.md)
- [X] Logging code meets the guideline
- [X] User guide and development docs updated (if needed)

### CODE REVIEW

File: pkg/pki/k8scsrpki/k8scsr_user.go [hunk 1]
```diff
@@ -153,10 +153,15 @@ func (c *k8sCSR) ReconcileUserCertificate(
 	}
 
 	//Leaf cert
-	secret.Data[corev1.TLSCertKey] = certs[len(certs)-1].ToPEM()
-	//Root CA cert
-	if certs[0].Certificate.IsCA {
-		secret.Data[v1alpha1.CoreCACertKey] = certs[0].ToPEM()
+	secret.Data[corev1.TLSCertKey] = certs[0].ToPEM()
```
Comment (chrisgacsal) [inline pkg/pki/k8scsrpki/k8scsr_user.go:156]:
```suggestion
var caChain []byte
for key, cr := range certs {
	if cr.Certificate.IsCA {
		caChain = append(caChain, cr.ToPEM()...)
	}
}
secret.Data[v1alpha1.CoreCACertKey] = append(caChain, byte('\n'))
```

File: pkg/pki/k8scsrpki/k8scsr_user.go [hunk 2]
```diff
@@ -153,10 +153,15 @@ func (c *k8sCSR) ReconcileUserCertificate(
...
+	//CA chain certs
+	for key, cr := range certs {
```
Comment (chrisgacsal) [inline pkg/pki/k8scsrpki/k8scsr_user.go:158]:
```suggestion
var caChain []byte
for key, cr := range certs {
	if cr.Certificate.IsCA {
		caChain = append(caChain, cr.ToPEM()...)
	}
}
secret.Data[v1alpha1.CoreCACertKey] = append(caChain, byte('\n'))
```

**Review by baluchicken — COMMENTED** [review]
I would use a different name than CoreCACertKey. using chain.pem should be fine.

**Review by chrisgacsal — APPROVED** [review]
LGTM! Thanks for fixing!

**Comment by baluchicken** [comment]
THX LGTM

**Review by adamantal — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**